server/app/services/DateConverter.java

```diff
@@ -32,14 +32,24 @@ public String formatIso8601Date(Instant instant) {
     return DateTimeFormatter.ISO_DATE.withZone(zoneId).format(instant);
   }
 
+  /**
+   * Converts a DateString in "YYYY-MM-dd" format to LocalDate type Throws DateTimeParseException
+   */
+  public LocalDate parseStringToLocalDate(String dateString) {
+    return LocalDate.parse(dateString, DateTimeFormatter.ofPattern("yyyy-MM-dd"));
```

Comment:
suggestion (blocking): please use the built-in format:
```suggestion
    return LocalDate.parse(dateString, DateTimeFormatter.ISO_DATE);
```